readthedocs/proxito/README.rst

```diff
@@ -42,6 +42,69 @@ path prefix:
                                     version     |
                                              filename
 
+Custom path prefixes
+--------------------
+
+By default we serve documentation from the following paths:
+
+- Multi version project: ``/<language>/<version>/<filename>``
+- Single version project: ``/<filename>``
+- Subproject (multi version): ``/projects/<subproject-alias>/<language>/<version>/<filename>``
+- Subproject (single version): ``/projects/<subproject-alias>/<filename>``
+
+Custom path prefixes can be used to change the prefix from where the documentation is served from,
+and even change the ``/projects`` prefix for subprojects.
+These prefixes can be changed per project, with the following Project model attributes:
+
+custom_prefix:
+   Add a prefix for multi version and single version projects.
+
+custom_subproject_prefix:
+   Change the ``/projects`` prefix for subprojects.
+   To change the prefix of the subproject itself, use ``custom_prefix`` on the subproject.
+
+Where to define the custom path prefix
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To change the path prefix of a project,
+define the prefix in the ``custom_prefix`` attribute of the project itself.
+For a translation, change the main language project ``custom_prefix`` attribute.
+
+And to change the prefix of subprojects (``/projects``),
+define the prefix in the ``custom_subproject_prefix`` attribute of the super project.
+
+We have validations in place to ensure that the custom prefix is defined in the correct project
+(this validations are run when the project is saved from a form or the admin).
+
+Examples
+~~~~~~~~
+
+Say we have the following projects:
+
+- docs (main project)
+- docs-es (spanish translation of the docs project)
+- subproject (subproject of the docs project)
+- subproject-es (spanish translation of the subproject project)
+
+They are normally served from:
+
+- docs.rtd.io/en/latest/
+- docs.rtd.io/es/latest/
+- docs.rtd.io/projects/subproject/en/latest/
+- docs.rtd.io/projects/subproject/es/latest/
+
+Then we add a custom path prefix like:
+
+- docs with a custom path prefix of ``/prefix/``
+- docs with a custom subproject path of ``/s/``
```

Comment:
Good question, they can overlap, but the path needs to be different from a language if they aren't a single version project, for example:

- /prefix/s/: this is fine, since we can distinguish `/prefix/s/project/en/latest/` from `/prefix/en/latest/`
- /prefix/es/: this isn't okay, since we won't be able to distinguish `/prefix/es/latest/` from `/prefix/es/es/latest/`
- /prefix/foo/es/: this is okay, since `foo` doesn't match a language.

Will update the docs and add some tests about this.